kvstore panics on empty map

<!--
Pro-tip: You can leave this block commented, and it still works!

Select the appropriate areas for your issue:

/area API

Classify what kind of issue this is:

/kind bug

-->

## Expected Behavior
Setting a key on an empty kvstore, i.e. ConfigMap with no data field, should not panic.

## Actual Behavior

```bash
panic: assignment to entry in nil map [recovered]
	panic: assignment to entry in nil map
```

## Steps to Reproduce the Problem

1. Create empty ConfigMap (name/namespace only)
2. Use kvstore
3. Set key -> panic

## Additional Info

--- a/kvstore/kvstore_cm.go
+++ b/kvstore/kvstore_cm.go
@@ -103,6 +103,9 @@ func (cs *configMapKVStore) Set(ctx context.Context, key string, value interface
 	if err != nil {
 		return fmt.Errorf("failed to Marshal: %w", err)
 	}
+	if cs.data == nil {
+		cs.data = map[string]string{}
+	}
 	cs.data[key] = string(bytes)
 	return nil
 }

--- a/kvstore/kvstore_cm_test.go
+++ b/kvstore/kvstore_cm_test.go
@@ -131,6 +131,43 @@ func TestLoadSaveUpdate(t *testing.T) {
 	if err != nil {
 		t.Error("failed to return string:", err)
 	}
+	if ret != "otherbar" {
+		t.Errorf("got back unexpected value, wanted %q got %q", "bar", ret)
+	}
+}
+
+func TestLoadSaveUpdateEmptyMap(t *testing.T) {
+	// empty map, i.e. no data field
+	cm := corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "ConfigMap",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+	}
+
+	tc := newTestClient([]runtime.Object{&cm}...)
+	cs := NewConfigMapKVStore(context.Background(), name, namespace, tc.clientset)
+	if err := cs.Init(context.Background()); err != nil {
+		t.Error("Failed to Init ConfigStore:", err)
+	}
+
+	if err := cs.Set(context.Background(), "jimmy", "otherbar"); err != nil {
+		t.Error("Failed to set key:", err)
+	}
+
+	if err := cs.Save(context.Background()); err != nil {
+		t.Error("Failed to save configmap:", err)
+	}
+
+	if tc.updated == nil {
+		t.Errorf("ConfigMap Not updated")
+	}
+	var ret string
+	err := cs.Get(context.Background(), "jimmy", &ret)
 	if err != nil {
 		t.Error("failed to return string:", err)
 	}
@@ -162,9 +199,6 @@ func TestLoadSaveUpdateComplex(t *testing.T) {
 	}
 	var ret testStruct
 	err = cs.Get(context.Background(), "jimmy", &ret)
-	if err != nil {
-		t.Error("failed to return string:", err)
-	}
 	if err != nil {
 		t.Error("failed to return string:", err)
 	}

